### DISCUSSION TITLE
Anyone having problems with preview of Markdown files?

### DISCUSSION THREAD

**Discussion by nzyeehaa** [datenstrom/community#682] [Ask a question] [answered]
Following a long day I encountered a problem after installing the search extension on a new development site...

There was **no** problem with the search extension, and It should have been simple, but because it was late and I had run out of steam, it took some time to realise the problem.

I prefer to use .txt extension for content rather then .md, so in the >yellow-system.ini, I have...

`CoreContentExtension: .txt`
`CoreContentDefaultFile: page.txt`
`CoreContentErrorFile: page-error-(.*).txt`
`EditNewFile: page-new-(.*).txt`

however, the search extension installs as _**/search/page.md**_ and the page is not found.

Yellow works out of the box, with its .md settings and extra settings make things much more complex. 
Each extension should **not** have to check for `CoreContentExtension:` and adapt its installation to suit, it would make extension development much more complex than it needs to be..
 ...however...
... Is it a _good_ idea for yellow to serve .md by default **AND** whatever is set at _CoreContentExtension:_?
...Or maybe, accept a comma separated list? 
...(both may be problematic if two files exist - page.md and page.txt)

Reasons I use .txt : 
Speed!
.md files arent native so .txt means I can use _any_ explorer/previewer to view each files _total_ content instead of _opening_ each file in an editor capable of .md. 
I have a powertoys installed so I _can_ preview .md files(without opening), but it renders them and trims the front matter, which is often what I am aiming to look at - therefore I still have to manually open every file in an editor.

**Comment by IngoBelka** [comment, marked as answer]
> md files arent native so .txt means I can use _any_ explorer/previewer

If you register filetype .md to open with the EDITOR - for example - as standard, you should get a preview in Explorer. There are a lot of tutorials in the web, how to realize that.  May be, you must activate the corresponding preview handler in Explorer.

**Reply by markseuffert** [reply]
It's similar on a Mac. You register md files with the standard text editor and get this preview:

![Screenshot md files](https://user-images.githubusercontent.com/2429165/135327657-25b1232a-2d01-4949-b01a-a1931dadf916.png)

**Reply by nzyeehaa** [reply]
@IngoBelka thanks for your comment.
msoft doesnt have a native .md previewer. 
I use atom editor and .md filetype is registered to open with atom...but preview in windows file explorer still doesnt work, the file has to be opened.
- (Atom has an awesome preview aspect without actually opening every file, but I regularly have several windows open and need to browse using file explorer)

I have Powertoys installed which gives a .md preview in file explorer - but it _renders_ the markdown so you miss out on the front matter.
Only other way to achieve it would be to hack the registry and fool msoft into opening .md as text...

...But back to yellow, its feature CoreContentExtension, and the question...

> Is it a good idea for yellow to serve .md by default **AND** whatever is _set_ at CoreContentExtension:?

..or is it simpler to remove the CoreContentExtension option and only serve .md

**Reply by nzyeehaa** [reply]
@markseu 
thought that may be the case, (but it was just an aside to the actual question)
on windows 10...
![yellow](https://user-images.githubusercontent.com/45002354/135382398-fe4ffac0-a90a-40c2-8056-7f6c5b1c235b.jpg)

**Reply by markseuffert** [reply]
On Windows 10 you have to configure the preview handler for `.md` files in Explorer:

1. Start `regedit` and navigate to key `HKEY_LOCAL_MACHINE\SOFTWARE\Classes\.md`
   on some computers it can be found at `HKEY_CURRENT_USER\SOFTWARE\Classes\.md`
2. Add a new string with `Name:PerceivedType`, `Type:REG_SZ`, `Data:text`
3. Have fun editing Markdown, you can also [preview other file types as text](https://superuser.com/questions/91804/windows-7-preview-other-file-types-as-text-in-preview-pane)
